scripts/check_branch.py

```diff
@@ -37,29 +36,36 @@ def __init__(self,
                          f'only {allowed}personal deployments.')
 
 
-def check_branch(branch: Optional[str], deployment: str) -> None:
+def check_branch(branch: str | None, deployment: str) -> None:
     deployment = config.Deployment(deployment)
     if deployment.is_shared:
         deployments = config.shared_deployments_for_branch(branch)
         if deployments is None or deployment not in deployments:
             raise BranchDeploymentMismatch(branch, deployment, deployments)
 
 
-def gitlab_branch() -> Optional[str]:
+def branch_context() -> str | None:
     """
-    Return the current branch if we're on GitLab, else `None`
-    """
-    # Gitlab checks out a specific commit which results in a detached HEAD
-    # (no active branch). Extract the branch name from the runner environment.
-    return os.environ.get('CI_COMMIT_REF_NAME')
-
-
-def local_branch() -> Optional[str]:
-    """
-    Return `None` if detached head, else the current branch
+    Determine the current branch context by checking multiple environment
+    variables across different CI platforms and local environments.
     """
+    gh_base_branch = os.environ.get('GITHUB_BASE_REF')
+    gh_feature_branch = os.environ.get('GITHUB_HEAD_REF')
+    gitlab_branch = os.environ.get('CI_COMMIT_REF_NAME')
     repo = git.Repo(config.project_root)
-    return None if repo.head.is_detached else repo.active_branch.name
+    if gh_base_branch is not None:
+        return gh_base_branch
+    elif gh_feature_branch is not None:
+        return gh_feature_branch
+    elif gitlab_branch is not None:
+        # Gitlab checks out a specific commit which results in a detached HEAD
+        # (no active branch). So the branch name from the runner environment is
+        # used.
```

Comment:
This comment is oddly placed. GitHub does the same thing so the comment should be generalized.